xarray/core/coordinates.py

```diff
@@ -151,6 +151,12 @@ def to_dataset(self):
     def _update_coords(self, coords):
         from .dataset import calculate_dimensions
 
+        for key in coords:
```

Comment:
Is this not already checked by all the callers, e.g., in `merge_coords_without_align`?